cmd/wfx-viewer/output/mermaid/mermaid_test.go

```diff
@@ -0,0 +1,27 @@
+package mermaid
+
+/*
+ * SPDX-FileCopyrightText: 2024 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/siemens/wfx/workflow/dau"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenerate(t *testing.T) {
+	buf := new(bytes.Buffer)
+	gen := NewGenerator()
+	err := gen.Generate(buf, dau.DirectWorkflow())
+	require.NoError(t, err)
+	expected := "stateDiagram-v2\n    [*] --> INSTALL\n    INSTALL --> INSTALLING: CLIENT\n    INSTALL --> TERMINATED: CLIENT\n    INSTALLING --> INSTALLING: CLIENT\n    INSTALLING --> TERMINATED: CLIENT\n    INSTALLING --> INSTALLED: CLIENT\n    INSTALLED --> ACTIVATE: WFX\n    ACTIVATE --> ACTIVATING: CLIENT\n    ACTIVATE --> TERMINATED: CLIENT\n    ACTIVATING --> ACTIVATING: CLIENT\n    ACTIVATING --> TERMINATED: CLIENT\n    ACTIVATING --> ACTIVATED: CLIENT\n    ACTIVATED --> [*]\n    TERMINATED --> [*]\n    classDef cl_INSTALL color:black,fill:#00cc00\n    class INSTALL cl_INSTALL\n    classDef cl_INSTALLING color:black,fill:#00cc00\n    class INSTALLING cl_INSTALLING\n    classDef cl_INSTALLED color:black,fill:#00cc00\n    class INSTALLED cl_INSTALLED\n    classDef cl_ACTIVATE color:black,fill:#00cc00\n    class ACTIVATE cl_ACTIVATE\n    classDef cl_ACTIVATING color:black,fill:#00cc00\n    class ACTIVATING cl_ACTIVATING\n    classDef cl_ACTIVATED color:black,fill:#4993dd\n    class ACTIVATED cl_ACTIVATED\n    classDef cl_TERMINATED color:black,fill:#9393dd\n    class TERMINATED cl_TERMINATED\n    Note right of INSTALL: <b>Group to Color Mapping</b><br/><font color=\"#00cc00\">OPEN</font> - regular workflow-advancing states<br/><font color=\"#4993dd\">CLOSED</font> - a successful update's terminal states<br/><font color=\"#9393dd\">FAILED</font> - a failed update's terminal states"
```

Comment:
to make this test readable, it could perhaps be made into an [approval test](http://approvaltests.com) via https://github.com/approvals/go-approval-tests

this will allow an easier adjustment of the plainly readable and formatted expected result

Reply:
Thanks for the hint, I quite like it and rewrote the tests to use this framework.